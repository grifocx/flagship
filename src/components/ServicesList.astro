---
import { getCollection } from 'astro:content';
import { ServicesList } from './ServicesList';

const services = await getCollection('services');
---

<ServicesList services={services.map(s => s.data)} />
