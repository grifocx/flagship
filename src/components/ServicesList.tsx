import React from 'react';
import { Card, CardContent, CardHeader, CardTitle } from './ui/card';
import { Wrench, Ruler, Calendar, Settings, Bike, PenTool as Tool, Cog, Battery, Monitor } from 'lucide-react';

const iconMap = {
  Wrench,
  Ruler,
  Calendar,
  Settings,
  Bike,
  Tool,
  Cog,
  Battery,
  Monitor,
};

interface Service {
  title: string;
  price: string;
  description: string;
  duration: string;
  icon: keyof typeof iconMap;
}

interface ServicesListProps {
  services: Service[];
}

export function ServicesList({ services }: ServicesListProps) {
  return (
    <div>
      <h2 className="text-2xl font-bold mb-6">Quality and Affordable Bicycle Services in Haymarket, VA</h2>
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6 mb-24">
        {services.map((service) => {
          const IconComponent = iconMap[service.icon];
          return (
            <Card key={service.title} className="hover-scale">
              <CardHeader>
                <div className="mb-4 inline-block p-3 bg-accent/10 rounded-lg">
                  <IconComponent className="h-6 w-6 text-accent" />
                </div>
                <CardTitle>{service.title}</CardTitle>
              </CardHeader>
              <CardContent>
                <p className="text-muted-foreground mb-4">{service.description}</p>
                <div className="flex items-center justify-between text-sm">
                  <span className="text-muted-foreground">Duration:</span>
                  <span>{service.duration}</span>
                </div>
                <div className="mt-4 text-lg font-semibold text-accent text-center">
                  {service.price}
                </div>
              </CardContent>
            </Card>
          );
        })}
      </div>
    </div>
  );
}