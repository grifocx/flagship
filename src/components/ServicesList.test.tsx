import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { ServicesList } from './ServicesList';

describe('ServicesList', () => {
  it('renders correctly', () => {
    const services = [
      {
        title: 'Basic Tune-Up - Level 1',
        price: '$99.99',
        description: 'Complete safety check, gear adjustment, brake adjustment, tire inspection, and basic cleaning.',
        duration: '1-2 days',
        icon: 'Wrench',
      },
      {
        title: 'Comprehensive Tune-Up - Level 2',
        price: '$159.99',
        description: 'Everything in basic tune-up plus bearing adjustments, wheel truing, and deep cleaning.',
        duration: '2-3 days',
        icon: 'Settings',
      },
    ];

    const { asFragment } = render(<ServicesList services={services} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
