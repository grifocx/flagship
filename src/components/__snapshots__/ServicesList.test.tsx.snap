// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServicesList > renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="text-2xl font-bold mb-6"
    >
      Quality and Affordable Bicycle Services in Haymarket, VA
    </h2>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6 mb-24"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm hover-scale"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <div
            class="mb-4 inline-block p-3 bg-accent/10 rounded-lg"
          >
            <svg
              class="lucide lucide-wrench h-6 w-6 text-accent"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.7 6.3a1 1 0 0 0 0 1.4l1.6 1.6a1 1 0 0 0 1.4 0l3.77-3.77a6 6 0 0 1-7.94 7.94l-6.91 6.91a2.12 2.12 0 0 1-3-3l6.91-6.91a6 6 0 0 1 7.94-7.94l-3.76 3.76z"
              />
            </svg>
          </div>
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Basic Tune-Up - Level 1
          </h3>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            class="text-muted-foreground mb-4"
          >
            Complete safety check, gear adjustment, brake adjustment, tire inspection, and basic cleaning.
          </p>
          <div
            class="flex items-center justify-between text-sm"
          >
            <span
              class="text-muted-foreground"
            >
              Duration:
            </span>
            <span>
              1-2 days
            </span>
          </div>
          <div
            class="mt-4 text-lg font-semibold text-accent text-center"
          >
            $99.99
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm hover-scale"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <div
            class="mb-4 inline-block p-3 bg-accent/10 rounded-lg"
          >
            <svg
              class="lucide lucide-settings h-6 w-6 text-accent"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
              />
              <circle
                cx="12"
                cy="12"
                r="3"
              />
            </svg>
          </div>
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            Comprehensive Tune-Up - Level 2
          </h3>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            class="text-muted-foreground mb-4"
          >
            Everything in basic tune-up plus bearing adjustments, wheel truing, and deep cleaning.
          </p>
          <div
            class="flex items-center justify-between text-sm"
          >
            <span
              class="text-muted-foreground"
            >
              Duration:
            </span>
            <span>
              2-3 days
            </span>
          </div>
          <div
            class="mt-4 text-lg font-semibold text-accent text-center"
          >
            $159.99
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
